actgen/agents/dqn.py

```diff
@@ -74,16 +80,19 @@ def update(self):
             # hard copy
             if self.n_training_steps % self.params['target_copy_period'] == 0:
                 self.q_target.hard_copy_from(self.q)
+        
+        # update inverse model
+        if self.params['inv_model']:
+            self.inverse_predictor.update(batch, encoder=self.q.encoder)
 
         return loss.detach().item()
 
     def _get_epsilon(self, testing=False):
         if testing:
             epsilon = self.params['epsilon_during_eval']
         else:
-            alpha = (len(self.replay) - self.params['replay_warmup_steps']) / self.params['epsilon_decay_period']
-            alpha = np.clip(alpha, 0, 1)
-            epsilon = self.params['epsilon_final'] * alpha + 1 * (1 - alpha)
+            epsilon = (self.params['epsilon_final'] + (1 - self.params['epsilon_final'])
+                 * math.exp(-1. * self.n_training_steps / self.params['epsilon_decay_period']))
```

Comment:
This was the switch to an exponential epsilon schedule, right?

Reply:
yes!